spec/models/product_spec.rb

```diff
@@ -0,0 +1,142 @@
+require "rails_helper"
+
+RSpec.describe Product, type: :model do
+  subject {product}
+
+  let!(:supplier) { FactoryBot.create :supplier }
```

Comment:
let! khác gì let e ?

Reply:
hoàn toàn có thể dùng let! thay let. khi khai báo đối tượng nào đấy nếu dùng let thì khi  trong 1 khối if (example) nếu gọi đến biến let nhiều hơn 1 lần thì cho dù có thay đổi giá trị của nó, thì gọi  biến đó ra nó chỉ có giá trị ban đầu. chỉ thay đổi giá trị khi sang example khacs( khối if khacs) . còn let! thì luôn luôn khởi tạo lại khi gọi biến đấy. nên dùng let! khi khai báo object sử dụng trong nhiều chỗ khi muốn gọi.

Reply:
let! giữa các khối it nó thay đổi giá trị k e

Reply:
Có chị đây là điểm khác biệt với let.